piet/src/gradient.rs

```diff
@@ -262,26 +288,52 @@ impl LinearGradient {
             stops: stops.to_vec(),
         }
     }
+
+    // maybe these should be public API? that was my original intention but I'm not
+    // sure there's a clear use, so keeping them private for now.
+    /// Generate a [`FixedLinearGradient`] by mapping points in the unit square
+    /// onto points in `rect`.
+    ///
+    /// [`FixedLinearGradient`]: struct.FixedLinearGradient.html
+    fn resolve(&self, rect: Rect) -> FixedLinearGradient {
+        FixedLinearGradient {
+            start: self.start.resolve(rect),
+            end: self.end.resolve(rect),
+            stops: self.stops.clone(),
+        }
+    }
 }
 
 impl RadialGradient {
-    /// Creates a simple `RadialGradient`. This gradient has the same `origin`
-    /// and `center`, and uses the `Fill` [`ScaleMode`]. These attributes can be
-    /// modified with the [`with_origin`] and [`with_scale_mode`] builder methods.
+    /// Creates a simple `RadialGradient`. This gradient has `origin` and `center`
```

Comment:
I think a short explanation of origin and center would be handy, as I don't think they're universally understood terms. Here's what I suggest:

The `center` parameter specifies the center of the circle, so that points of distance `radius` from the center map to 1.0 in the gradient stops. The `origin` parameter specifies the point that maps to 0.0 in the gradient stops. Using the same origin and center gives a radially symmetric gradient; using different points is useful for simulating lighting effects.

Reply:
Thanks, I struggled to communicate that distinction clearly.

Reply:
although I'll put this in the type-level docstring, since neither of these fields are exposed in `new`.